src/pages/TriggerReleasePlanPage.tsx

```diff
@@ -1,20 +1,66 @@
 import * as React from 'react';
 import { Helmet } from 'react-helmet';
+import { useParams } from 'react-router-dom';
+import { useK8sWatchResource } from '@openshift/dynamic-plugin-sdk-utils';
+import { Bullseye, Spinner } from '@patternfly/react-core';
 import NamespacedPage from '../components/NamespacedPage/NamespacedPage';
 import PageAccessCheck from '../components/PageAccess/PageAccessCheck';
 import { TriggerReleaseFormPage } from '../components/ReleaseService/ReleasePlan/TriggerRelease/TriggerReleaseFormPage';
 import { FULL_APPLICATION_TITLE } from '../consts/labels';
-import { ReleaseModel } from '../models';
+import { ReleaseModel, ReleasePlanGroupVersionKind } from '../models';
+import ErrorEmptyState from '../shared/components/empty-state/ErrorEmptyState';
+import { HttpError } from '../shared/utils/error/http-error';
 import { AccessReviewResources } from '../types';
+import { ReleasePlanKind } from '../types/coreBuildService';
+import { useWorkspaceInfo } from '../utils/workspace-context-utils';
 
 const TriggerReleasePlanPage: React.FC<React.PropsWithChildren<unknown>> = () => {
+  const { name } = useParams();
+  const { namespace } = useWorkspaceInfo();
+
+  const [releasePlan, loaded, loadErr] = useK8sWatchResource<ReleasePlanKind>({
+    groupVersionKind: ReleasePlanGroupVersionKind,
+    isList: false,
+    name,
+    namespace,
+  });
+
   const accessReviewResources: AccessReviewResources = [{ model: ReleaseModel, verb: 'create' }];
 
+  if (!name) {
```

Comment:
you should check for the `releasePlan` here to see if it exists. If `name` is not present in URL, you should show the form without release plan preselected so the user can fill out the form.

Reply:
updated